### DISCUSSION TITLE
Routing on own Bundle

### DISCUSSION THREAD

**Discussion by aguerer-remind** [pimcore/pimcore#16640] [General]
Hi guys,

I have created my own bundle but it fails on routing. Im using Pimcore 11.1

my bundle has an settings.js like that

```
pimcore.registerNS("pimcore.plugin.ContactsBundle.settings");

pimcore.plugin.ContactsBundle.settings = Class.create({

    activate: function () {
        var tabPanel = Ext.getCmp('pimcore_panel_tabs');
        
        if(tabPanel.items.keys.indexOf("contactsbundle_settings") > -1) {
            tabPanel.setActiveItem('contactsbundle_settings');
        } else{
            throw "need to reload";
        }
    },

    initialize: function () {
        this.getData();
    },

    getData: function () {
        Ext.Ajax.request({
            url: '/admin/contactsmenu/get-settings-attributes',
            success: function (response) {
                console.log(response);
            }.bind(this)
        });
    },
});
```

under src/Controller/admin/ContactsMenuController.php i have this controller class:

```
namespace Remind\ContactsBundle\Controller\Admin;


use Symfony\Component\HttpFoundation\Request;
use Pimcore\Bundle\AdminBundle\Controller\AdminAbstractController;
use Symfony\Component\Routing\Annotation\Route;

/**
 * @Route("/admin/contactsmenu")
 */
class ContactsMenuController extends AdminAbstractController
{
    /**
     * @Route("/get-settings-attributes")
     */
    public function getSettingsAttributesAction(Request $request)
    {
        $mm2AwgValues = [];

        $response = [
            "mm2AwgValues" => $mm2AwgValues,
        ];

        return $this->adminJson(["data" => $response]);
    }
}
```

if i click to my menu i get error with no route found
<img width="753" alt="Bildschirmfoto 2024-02-20 um 10 55 33" src="https://github.com/pimcore/pimcore/assets/154238908/81deeec3-81cd-4000-9954-3cab257e9b21">

whats wrong here.

**Comment by jdreesen** [comment]
You need to “expose” your routes to be able to use them in ExtJS: https://github.com/FriendsOfSymfony/FOSJsRoutingBundle/blob/master/Resources/doc/usage.rst#generating-uris

**Reply by aguerer-remind** [reply]
Hi, thanks for answer.

im added now on config/routing.yaml this configuration:

```
admin_contactsmenu:
    path: /admin/contactsmenu/get-settings-attributes
    defaults: { _controller: Remind\ContactsBundle\Controller\Admin\ContactsmenuController::getSettingsAttributes }
```

but problem still exists...

**Reply by jdreesen** [reply]
But did you add the “expose” flag?

Either in your `config/routing.yaml`:
```yaml
admin_contactsmenu:
	...
    options:
        expose: true
```

or in your `@Route` annotations:

```php
    /**
     * @Route("/get-settings-attributes", options={"expose"=true})
     */
```

**Reply by aguerer-remind** [reply]
my file structure is that

<img width="493" alt="Bildschirmfoto 2024-02-20 um 11 54 16" src="https://github.com/pimcore/pimcore/assets/154238908/39873936-707c-4589-8eff-6be58467b33b">

my routing.yaml includes that:

```
admin_contactmenu:
    path:     /admin/contact-menu/get-settings-attributes
    defaults: { _controller: ContactsBundle\Controller\Admin\ContactsMenuController:getSettingsAttributes }
    options:
        expose: true
```

and thats now my ContactsMenuController:

```
<?php
namespace Remind\ContactsBundle\Controller\Admin;


use Symfony\Component\HttpFoundation\Request;
use Pimcore\Bundle\AdminBundle\Controller\AdminAbstractController;
use Symfony\Component\Routing\Annotation\Route;

/**
 * @Route("/admin/contact-menu")
 */
class ContactsMenuController extends AdminAbstractController
{
    /**
     * @Route("/get-settings-attributes", options={"expose"=true})
     */
    public function getSettingsAttributesAction(Request $request)
    {
        $mm2AwgValues = [];

        $response = [
            "mm2AwgValues" => $mm2AwgValues,
        ];

        return $this->adminJson(["data" => $response]);
    }
}
```

**Reply by aguerer-remind** [reply]
I have now executed the following command

bin/console fos:js-routing:dump

and tada my route existing now 

<img width="1225" alt="Bildschirmfoto 2024-02-20 um 12 56 47" src="https://github.com/pimcore/pimcore/assets/154238908/ba922fe1-6761-40e8-9138-5330d5c21138">


if i click now to my button i get these error

<img width="934" alt="Bildschirmfoto 2024-02-20 um 12 58 26" src="https://github.com/pimcore/pimcore/assets/154238908/ac1e4a7f-3534-4ba0-b4d5-c3e25371e67d">

whats wrong now?
